src/components/ModelizerModelView.vue

```diff
@@ -141,14 +165,51 @@ async function updatePluginsAndTemplates() {
       templates.value = response;
     })
     .catch(() => {
-      Notify.create({
-        type: 'negative',
-        message: t('errors.templates.getData'),
-        html: true,
-      });
+      notifyError();
     });
 }
 
+/**
+ * Instantiate from a dragged component definition or template.
+ * @param {DragEvent} event - The drag event.
+ * @return {Promise<void>} - Implicitly declared Promise wrapping the function.
+ */
+async function dropHandler(event) {
+  const dropData = JSON.parse(event.dataTransfer.getData('text/plain'));
+  const modelFolder = process.env.MODELS_DEFAULT_FOLDER
+    ? `${process.env.MODELS_DEFAULT_FOLDER}/${route.query.path}`
+    : `${route.query.path}`;
+
+  let files;
+
+  if (dropData.isTemplate) {
+    const activeTemplate = templates.value.find(
+      ({ key }) => key === dropData.definitionType,
+    );
+
+    files = await renderModel(route.params.projectName, modelFolder, data.plugin);
+
+    await Promise.all(activeTemplate.files
+      .map((file) => getTemplateFileByPath(`templates/${activeTemplate.key}/${file}`)
+        .then((result) => appendProjectFile(route.params.projectName, new FileInput({
+          path: `${modelFolder}/${file}`,
+          content: generateTemplate(result.data),
+        })))
+        .catch(() => {
+          notifyError();
+        })));
+  } else {
+    const newComponentDefinition = data.plugin
+      .data.definitions.components
+      .find((def) => def.type === dropData.definitionType);
```

Comment:
`.find(({ type }) => type === dropData.definitionType);`
Can you put in in the line before ?

Reply:
Erratum : merge l.202 + l.203 (not l.203 + l.204)

Reply:
done